tools/vscode/ddlog-language/src/syntax/entity.json5

```diff
@@ -0,0 +1,84 @@
+{
+  "repository": {
+        "entity": {
+            "patterns": [
+                {
+                    "match": "(\\b)(?<=relation)(\\s)([A-Z0-9_][a-zA-Z0-9_]+)",
+                    "name": "entity.name.relation.ddlog"
+                },
+                {
+                    "match": "(\\b)(?<=relation)(\\s)([A-Z0-9_]*)",
+                    "name": "entity.name.relation.ddlog"
+                },                             
+                {
+                    "match": "\\b(?<=function)(\\b)([a-z_][a-zA-Z0-9_]*)",
+                    "name": "entity.name.function.ddlog",
+                    "captures": {
+                        "1": {
+                            "name": "keyword.operator.group.parameters.ddlog"
+                        }
+                    }
+                },
+                {
+                    "match": "(\\b)([a-z_][a-zA-Z0-9_]+)(?=[(])",
+                    "name": "entity.name.function.ddlog",
+                    "captures": {
+                        "1": {
+                            "name": "keyword.operator.group.parameters.ddlog"
+                        }
+                    }
+                },
+                {
+                    "match": "(\\b)([a-z_]+)(?=[(])",
+                    "name": "entity.name.function.ddlog",
+                    "captures": {
+                        "1": {
+                            "name": "keyword.operator.group.parameters.ddlog"
+                        }
+                    }
+                },
+                {
+                    "match": "(?<=')([A-Z][a-zA-Z0-9_]*)",
+                    "name": "entity.name.type-variable.ddlog"
+                },
+                {
+                    "match": "(?<=(\\=))(\\s*)([A-Z][a-zA-Z0-9_]*)",
+                    "name": "entity.name.constructor.ddlog"
+                },
+                {
+                    "match": "(?<=(\\|))(\\s*)([A-Z][a-zA-Z0-9_]*)",
+                    "name": "entity.name.constructor.ddlog"
+                },                                 
+                {
+                    "match": "(\\b)(?<=import)(\\b)([a-z_][a-zA-Z0-9_]*)",
+                    "name": "entity.name.import_decl.ddlog"
+                },
+                {
+                    "match": "\\b(?<=typedef)\\s*([a-zA-Z_][a-zA-Z0-9_]*)",
+                    "name": "entity.name.type.ddlog"
+
+                }, 
+                {
+                    "match": "\\b(?<=type)\\s*([a-zA-Z_][a-zA-Z0-9_]*)",
+                    "name": "entity.name.type.ddlog"
+
+                },
+                {
+                    "begin": "\\b(typedef)\\s*([a-zA-Z_][a-zA-Z0-9_]*)(=)([A-Z][a-zA-Z0-9_]+)",
```

Comment:
after = you have a type_spec in the ddlog grammar, which is not just an identifier as in your description here. Also, 
there are no optional spaces in your regex.